packages/medusa/src/interfaces/price-selection-strategy.ts

```diff
@@ -29,25 +158,114 @@ export interface IPriceSelectionStrategy extends ITransactionBaseService {
+   *
+   * @param {string[]} variantIds - The IDs of the updated variants.
+   * @returns {Promise<void>} Resolves after any necessary actions are performed.
+   *
+   * @example
+   * For example, this is how this method is implemented in the Medusa backend's default
+   * price selection strategy:
+   *
+   * ```ts
+   * import {
+   *   AbstractPriceSelectionStrategy,
+   *   CustomerService,
+   * } from "@medusajs/medusa"
+   * import { promiseAll } from "@medusajs/utils"
+   *
+   * type InjectedDependencies = {
+   *   customerService: CustomerService
+   * }
+   *
+   * export default class MyStrategy extends
+   *   AbstractPriceSelectionStrategy {
+   *
+   *   public async onVariantsPricesUpdate(variantIds: string[]): Promise<void> {
+   *     await promiseAll(
+   *       variantIds.map(
+   *         async (id: string) => await this.cacheService_.invalidate(`ps:${id}:*`)
+   *       )
+   *     )
+   *   }
+   *
+   *   // ...
+   * }
+   * ```
+   *
+   * :::note
+   *
+   * Learn more about the cache service in [this documentation](https://docs.medusajs.com/development/cache/overview).
+   *
+   * :::
    */
   onVariantsPricesUpdate(variantIds: string[]): Promise<void>
 }
 
+/**
+ * @parentIgnore activeManager_,atomicPhase_,shouldRetryTransaction_,withTransaction
+ */
 export abstract class AbstractPriceSelectionStrategy
   extends TransactionBaseService
   implements IPriceSelectionStrategy
 {
+  /**
+   * @ignore
+   */
   static _isPriceSelectionStrategy = true
 
+  /**
+   * @ignore
+   */
   static isPriceSelectionStrategy(object): boolean {
     return object?.constructor?._isPriceSelectionStrategy
   }
 
+  /**
+   * You can use the `constructor` of your price-selection strategy to access the different services in Medusa through dependency injection.
+   *
+   * @param {MedusaContainer} container - An instance of `MedusaContainer` that allows you to access other resources, such as services, in your Medusa backend.
+   * @param {Record<string, unknown>} config - If this price-selection strategy is created in a plugin, the plugin's options are passed in this parameter.
+   *
+   * @example
+   * // ...
+   * import {
+   *   AbstractPriceSelectionStrategy,
+   *   CustomerService,
+   * } from "@medusajs/medusa"
+   * type InjectedDependencies = {
+   *   customerService: CustomerService
+   * }
+   *
+   * class MyStrategy extends
+   *   AbstractPriceSelectionStrategy {
+   *
+   *   protected customerService_: CustomerService
+   *
+   *   constructor(container: InjectedDependencies) {
+   *     super(container)
+   *     this.customerService_ = container.customerService
+   *   }
+   *
+   *   // ...
+   * }
+   *
+   * export default  MyStrategy
+   */
+  protected constructor(
+    protected readonly container: MedusaContainer,
+    protected readonly config?: Record<string, unknown> // eslint-disable-next-line @typescript-eslint/no-empty-function
+  ) {
+    super(container, config)
+  }
+
```

Comment:
the introduction of the constructor (before this PR it was auto extended from TransactionBaseService which had container: any) broke the example written here above (and the one from the documentation). I don't know now how to inject dependencies as passing InjectedDependencies Object is not allowed by the type `MedusaContainer` now anymore.

Related issue: https://discord.com/channels/876835651130097704/1201534153896763472/1201534153896763472

Reply:
Thank you for pointing this out @tobiaskraus , I'll open another PR to fix it. In the meantime, you can instead use `super(arguments[0])` instead of `super(container)`